pyam/core.py

```diff
@@ -1565,6 +1537,76 @@ def map_regions(self, map_col, agg=None, copy_col=None, fname=None,
         if not inplace:
             return ret
 
+    def subtract(self, other, axis, new_name, ignore_meta_conflict=False):
```

Comment:
1) suggest to use `axis='variable'` as default

2) suggest to not have this operation with another `IamDataFrame`as input, but specify `a` and `b` directly to be used as filters (i.e, `a - b`).

Say you have an `IamDataFrame` with total CO2 emissions and energy-specific CO2 emissions, and you want to compute the difference and append it to the original data.

With the currently implemented API, you'll have
```
co2_total = df.filter(variable='Emissions|CO2')
co2_ene = df.filter(variable='Emissions|CO2|Energy')
diff = co2_total.subtract(co2_ene, axis='variable', name='Emissions|CO2|Other')
df.append(diff, inplace=True)
```

compared to
```
df.subtract(name='Emissions|CO2|Other', a='Emissions|CO2', b='Emissions|CO2|Energy',
            axis='variable', append=True)
```

Reply:
> * suggest to use `axis='variable'` as default
> *

don't disagree
> suggest to not have this operation with another IamDataFrameas input, but specify a and b directly to be used as filters (i.e, a - b).

Here, I do disagree. Our original goal with the `IamDataFrame` class was to avoid direct inheritance, but to make the interface as close to a `pd.DataFrame` as possible using composition (as suggested by [pandas devs themselves](https://pandas.pydata.org/pandas-docs/stable/development/extending.html#subclassing-pandas-data-structures).

The method signature as implemented closely mimics the [`pd.Dataframe.subtract()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.subtract.html) method signature, which i would suggest keeping.

I think a middle way solution here would be to also add methods like:

```
pyam.subtract(df, a=..., b=..., ...)
```

which behaves as you suggest

Reply:
I agree with the high-level principle to follow pandas, but we should also consider where our use cases diverge. And with maths operations, I think that within-`IamDataFrame` are a more common use case than comparing two instances.

In response to your middle-way suggestion, I'd think that switching the behaviour would make a lot more sense:
 - `IamDataFrame.subtract(name, a, b, axis)` for the within-dataframe operations
 - `pyam.substract(df1, df2)` for the across-dataframe operations

Reply:
It seems we then disagree here. The position I come from is that every operation function on an object I have ever seen or written myself comes in the form of `object.op(other, ...)`. 

Given that we

a) agree that in general operations are useful
b) disagree on the exact interface for those operations

I would suggest then that we provide here a warning that this is an experimental feature which is volatile and may change soon.